tracer/src/Datadog.Trace/SpanContextInjector.cs

```diff
@@ -65,31 +65,30 @@ public void InjectIncludingDsm<TCarrier>(TCarrier carrier, Action<TCarrier, stri
             InjectInternal(carrier, setter, context, messageType, target);
         }
 
-        internal static void InjectInternal<TCarrier>(TCarrier carrier, Action<TCarrier, string, string> setter, ISpanContext context, string? messageType = null, string? target = null)
+        internal static void InjectInternal<TCarrier>(TCarrier carrier, Action<TCarrier, string, string> setter, ISpanContext? context, string? messageType = null, string? target = null)
         {
             if (messageType != null && target == null) { ThrowHelper.ThrowArgumentNullException(nameof(target)); }
             else if (messageType == null && target != null) { ThrowHelper.ThrowArgumentNullException(nameof(messageType)); }
 
-            if (context == null!) { ThrowHelper.ThrowArgumentNullException(nameof(context)); }
+            var spanContext = context as SpanContext;
```

Comment:
Can we still throw if `ISpanContext? context` is null?

Reply:
This (and `SpanContextExtractor`) are weird because they are public APIs and we can't change the signature to inject/extract both `SpanContext` and `Baggage`, so they are acting on the static `Baggage.Current` instead.

Normally I would say "no," we _should_ allow `SpanContext` to be `null` because it allows the user to inject baggage only. But that seems weird given than it's called `SpanContextInjector` and there is no indication that we are magically injecting the static `Baggage.Current`.

So yeah, I can make this throw again if you still agree.

Reply:
Crazy thought, but what if we keep this method behaving the same as before, so that it _doesn't_ add Baggage? Then if users actually want to inject Baggage, then we add an overload that accepts Baggage as an argument. What do you think?